src/nnet3/nnet-am-decodable-simple.h

```diff
@@ -392,6 +392,170 @@ class DecodableAmNnetSimpleParallel: public DecodableInterface {
+     
+     Note: it stores references to all arguments to the constructor, so don't
+     delete them till this goes out of scope.
+
+     @param [in] opts   The options class.  Warning: it includes an acoustic
+                        weight, whose default is 0.1; you may sometimes want to
+                        change this to 1.0.
+     @param [in] nnet   The neural net that we're going to do the computation with
+     @param [in] priors Vector of priors-- if supplied and nonempty, we subtract
+                        the log of these priors from the nnet output.
+     @param [in] compiler  A pointer to the compiler object to use-- this enables the
+                        user to maintain a common object in the calling code that
+                        will cache computations across decodes.  Note: the compiler code
+                        has no locking mechanism (and it would be tricky to design one,
+                        as we'd need to lock the individual computations also),
+                        so the calling code has to make sure that if there are
+                        multiple threads, they do not share the same compiler
+                        object.
+     @param [in] online_ivector_period If you are using iVectors estimated 'online'
+                        (i.e. if online_ivectors != NULL) gives the periodicity
+                        (in frames) with which the iVectors are estimated.
+     @param [in] minibatch_size The capacity of the batch. In general, it means
+                        the number of chunks will be process.
+     @param [in] max_capacity The number of feature matrixes can be stored in
+                        this class. In the case of "very small batch size" or
+                        "super long utterance", it is very useful.
+  */
+  BatchNnetComputer(const NnetSimpleComputationOptions &opts,
+                    const Nnet &nnet,
+                    const VectorBase<BaseFloat> &priors,
+                    CachingOptimizingCompiler *compiler,
+                    int32 online_ivector_period = 0,
+                    int32 minibatch_size = 1,
+                    int32 max_capacity = 10);
+
+
+  // returns the number of frames of likelihoods.  The same as feats_.NumRows()
+  // in the normal case (but may be less if opts_.frame_subsampling_factor !=
+  // 1).
+  inline int32 NumFrames(std::string& utt_id) const { return *num_subsampled_frames_.find(utt_id); }
+
+  inline int32 OutputDim() const { return output_dim_; }
+
+  inline bool Ready() const { return batch_info_queue_.size() == minibatch_size_; }
+  
+  // The "very small batch size" or "super long utterance" case will cause the
+  // data is piled up. To prevent it, we lead in this function.
+  // When this function is triggered. We always call Compute() and 
+  // GetFinishedUtterance to reduce the load of the class.
+  inline bool Overload() const { return utt_list_.size() > max_capacity_; }
```

Comment:
I hope there is a way to get rid of the need for this function.  It seems rather ugly.